### ISSUE TITLE
How do you set the color of the main app window in a Windows host?

### ISSUE THREAD

**Issue by TheCakeMonster** [dotnet/MobileBlazorBindings#359]
I'm working with a MBB hybrid app that was created from the hybrid app template from MBB preview 6. The Windows app created has a blue window bar - the bar with the minimize/maximize controls and so on. I want to change the colour of this bar - it's an odd blue colour that doesn't seem to match anything else on Windows. However, despite many hours of trying, I can't work out how to change it.

![image](https://user-images.githubusercontent.com/46025270/114394817-8c395f00-9b93-11eb-97b9-c6161b8ac3da.png)

Could you possibly spare me a few minutes to point me in the right direction please?

In the live visual tree in Visual Studio, I can see that MainWindow has a nested BorderWindow control on which the BorderBrush has the colour applied. However, I can't find a way to walk the control tree in Xamarin Forms to modify this property. I tried investigating changing the control template but I can't suss that either; I don't know how to create a custom template.

The partial C# bootstrapping that is used in a MBB app is what appears to make this so difficult. If it was XAML, I think I could apply resources or something - but even that would be hard because I'm not in my comfort zone with the Xamarin part of the application.

I'm aware that this is really a Xamarin Forms question, but I suspect that trying to ask a question about MBB on any Xamarin forum would lead to a very confusing conversation, as they would talk to me about how to change the XAML, and there isn't any XAML in the app for me to change!

Thanks in advance.

**Comment by Eilon** [comment]
Hi @TheCakeMonster the Windows targets for MBB are based on WPF, so you could look for either a WPF-based solution of a Xamarin.Forms/WPF solution. I've done some searches and it seems like people have found ways to work around this directly in WPF, for example, https://stackoverflow.com/questions/1283006/changing-wpf-title-bar-background-color

**Comment by TheCakeMonster** [comment]
@Eilon thanks. After more searching and more attempts to understand walking the visual tree in Xamarin, I think the answer is that it looks to be very, very, very difficult to change. It's easy to change through Visual Studio's Live Visual Tree pane, so it is possible, but it's really hard for a novice to work out how Visual Studio does what it does.

It's really odd that this behaviour is so unlike the defaults for the target platform, and also so hard to change. I wouldn't want to do this if the colour chosen was at all close to any of those in the colour palette of the operating system, but as far as I can see, it isn't. It's just a hard-coded, and very unappealing, colour. It makes styling your app almost pointless, as any colours you might choose you use clash with the title bar as it's such a strong colour.

I think the team should expect people to be very disappointed by this when they try MAUI for themselves in .NET 6. Maybe this feedback needs to go in another repo. Can you think where I might post this that the Xamarin team will find it?

**Comment by TheCakeMonster** [comment]
Oh, OK, maybe not quite as hard as I thought!

You can manually add resources to the Resources collection of the application, and if they have the correct key, they are used to affect the appropriate items.

In the class that forms the entry point of the Windows application, adding these two lines changes the title bar from white on blue to black on white:

`			this.Resources.Add("CommandBarBackgroundColor", System.Windows.Media.Brushes.White);
			this.Resources.Add("CommandBarTextColor", System.Windows.Media.Brushes.Black);
`

**Comment by TheCakeMonster** [comment]
Question has been answered. It is possible to do this, although you need to know the trick of how to apply styling to Xamarin Forms without there being any XAML involved; it requires manipulation of the Resources collection on the MainWindow class (or whatever your implementation of FormsApplicationPage class is called)

**Comment by TheCakeMonster** [comment]
I've created an issue in the Xamarin Forms documentation repo about this topic.

https://github.com/xamarin/Xamarin.Forms-api-docs/issues/284

**Comment by Eilon** [comment]
Thank you @TheCakeMonster for following up! In .NET MAUI the main way to build Windows apps is based on WinUI, not WPF, so maybe this will be easier in .NET MAUI? Though, in .NET 6 you can also build Blazor hybrid apps directly in WPF and in WinForms.

**Comment by TheCakeMonster** [comment]
@Eilon ah, interesting to hear. Yes, I think that might make a significant difference. I think a WinUI app will automatically pick up its colours from the theme in use, which it appears that WPF cannot/does not.

If I understand the behaviour I was seeing WPF is based on the older style of colouring of Windows, where there was a lot of customisation available. In older version of Windows each colour was individually editable based upon its usage. It appears that later versions of Windows have dropped this in favour of theming, where individual colours do not seem to be editable. Metro-style apps, as they were called for a while, seem to respect the theme rather than the older colour palette.

In short, I think something based on WinUI will automatically look more in keeping with other apps, whereas WPF apps do not do this - at least not out of the box.

I'm all for the world domination of Blazor continuing, so these extra options sound good! Thanks for the information.
